crates/red_knot_python_semantic/resources/mdtest/comparison/byte_literals.md

```diff
@@ -0,0 +1,40 @@
+### Comparison: Byte literals
```

Comment:
nit: it would be nice to have maybe a sentence of documentation here that explains what the purpose of theses tests is. Even though for this file in particular it's kind of self-evident, I think it's useful to establish a convention here:

```suggestion
### Comparison: Byte literals

These tests assert that we infer precise `Literal` types for comparisons between objects
inferred as having `Literal` bytes types:
```